src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -21,9 +22,13 @@ public Task Start(IMessageSession session)
             var startableTasks = new List<Task>();
             foreach (var startable in wantToRunWhenBusStartsAndStops)
             {
-                var task = startable.Start(session).ThrowIfNull();
-
                 var startable1 = startable;
+                var startableName = startable1.GetType().AssemblyQualifiedName;
+
+                var longRunningTokenSource =
+                    LongRunningWarning($"The start method for {startableName} is taking too long to complete. The endpoint will not start until this operation has completed. Until then messages that arrive in the endpoint's queue will not be processed.");
```

Comment:
> "...is taking too long to complete."

Since that value is known, would it help if we specify that it takes a task longer than X (configured time) to complete?